### PR TITLE
Harmony 1061

### PR DESCRIPTION
**Pull request by hailiangzhang**
This PR will remove the granule limits for variable-based (e.g. Giovanni) service.

### CODE REVIEW

**Review by chris-durbin — CHANGES_REQUESTED** [review]
Looks good - only a few minor changes requested.

File: app/middleware/cmr-granule-locator.ts [hunk 1]
```diff
@@ -42,6 +42,8 @@ function getBbox(collection: cmr.CmrCollection, granule: cmr.CmrGranule): Boundi
 function getMaxGranules(req: HarmonyRequest): number {
   const query = keysToLowerCase(req.query);
 
+  if ( req.context.serviceConfig.has_granule_limit == false ) return Number.MAX_SAFE_INTEGER;
```
[the remark this answers is not in the document]
Comment (hailiangzhang) [reply app/middleware/cmr-granule-locator.ts]: Good catch ... changed now.

File: app/models/services/base-service.ts [hunk 2]
```diff
@@ -431,6 +433,8 @@ export default abstract class BaseService<ServiceParamType> {
       return operation.isSynchronous;
     }
 
+    if (this.config.force_sync !== undefined) return this.config.force_sync;
```
[the remark this answers is not in the document]
Comment (hailiangzhang) [reply app/models/services/base-service.ts:436]: Done ... thanks!

File: config/services.yml [hunk 3]
```diff
@@ -59,6 +59,8 @@ https://cmr.earthdata.nasa.gov:
   - name: gesdisc/giovanni
     data_operation_version: '0.12.0'
     enabled: !Env ${EXAMPLE_SERVICES}
+    has_granule_limit: false
```
[the remark this answers is not in the document]
Comment (hailiangzhang) [reply config/services.yml:62]: Just changed `force_sync` to `default_sync` ... will modify `docs/adapting-new-services.md`

[the remark this answers is not in the document]
Comment (hailiangzhang) [reply config/services.yml:62]: Done.

File: test/models/services.ts [hunk 4]
```diff
@@ -444,20 +447,53 @@ describe('services.chooseServiceConfig and services.buildService', function () {
...
-      const operation = new DataOperation();
-      operation.addSource(collectionId, [{ meta: { 'concept-id': variableId }, umm: { Name: 'the-var' } }]);
-      operation.outputFormat = 'text/csv';
+      let operation;
+
+      it('sets to synchronous for the variable-based-service', function () {
+        operation = new DataOperation();
+        operation.addSource(collectionId, [{ meta: { 'concept-id': variableId }, umm: { Name: 'the-var' } }]);
+        operation.outputFormat = 'text/csv';
+        new TurboService(this.config[0], operation);
```
[the remark this answers is not in the document]
Comment (hailiangzhang) [reply test/models/services.ts]: Yes, it's better to check for `service.isSynchronous` ... Done.

File: docs/adapting-new-services.md [hunk 5]
```diff
@@ -85,6 +85,8 @@ The structure of an entry in the [services.yml](../config/services.yml) file is
 ```yaml
 - name: harmony/service-example     # A unique identifier string for the service, conventionally <team>/<service>
   data_operation_version: '0.13.0' # The version of the data-operation messaging schema to use
+  has_granule_limit: true          # Optional flag indicating whether we will impose granule limts for the request. Default to true.
+  default_sync: false              # Optional flag indicating whether we will force the request to run synchrously. Default to false.
```
Comment (chris-durbin) [inline docs/adapting-new-services.md:89]: synchronously typo

**Review by chris-durbin — APPROVED** [review]

**Pull request merged**